File: NotifMsg/mainframe.cpp
#include "mainframe.h"

#include "main.h"

#ifndef __WXMSW__
#include "wxwin32x32.xpm"
#endif // __WXMSW__

#include <wx/notifmsg.h>

MainFrame::MainFrame(const wxString& title) : wxFrame(NULL, -1, title)
{
#ifdef __WXDEBUG__
	wxPrintf(_T("Creating a \"MainFrame\" object\n"));
#endif // __WXDEBUG__

	SetIcon(wxICON(appIcon)); // Defining app icon

	CreateControls();

	ConnectControls();
}

MainFrame::~MainFrame()
{
#ifdef __WXDEBUG__
	wxPrintf(_T("Destroying a \"MainFrame\" object\n"));
#endif // __WXDEBUG__
}

void MainFrame::CreateControls()
{
	// Status bar
	CreateStatusBar(1);
	wxString sTxt=wxGetApp().GetBuildInfos(true);
	SetStatusText(sTxt);

	// Add a panel to the frame (needed under Windows to have a nice frame)
	wxPanel *pnl=new wxPanel(this, -1);

	wxBoxSizer *szrMain=new wxBoxSizer(wxVERTICAL);
		szrMain->AddStretchSpacer(1);
		m_btnShowMsg=new wxButton(pnl, -1, _T("Show a notification message"));
		szrMain->Add(m_btnShowMsg, 0, wxALL|wxALIGN_CENTER, 5);
		szrMain->AddStretchSpacer(1);
	pnl->SetSizer(szrMain);
}

void MainFrame::ConnectControls()
{
	m_btnShowMsg->Bind(wxEVT_BUTTON, &MainFrame::OnBtnShowMsgClicked, this);
}

void MainFrame::OnBtnShowMsgClicked(wxCommandEvent &event)
{
	wxString sTitle = _T("wxWidgets notification");
	wxString sMsg = _T("This is a notification message.\n\nWelcome on wxWidgets ;-)");
	wxNotificationMessage nmsg(sTitle, sMsg);
	nmsg.SetFlags(wxICON_INFORMATION);

	nmsg.Show(wxNotificationMessage::Timeout_Auto);
}

File: NotifMsg/Python/MainFrame.py
import wx
import wx.adv

class MainFrame(wx.Frame):

    def __init__(self):
        wx.Frame.__init__(self, None, title=wx.GetApp().GetAppName())
        self.SetIcon(wx.Icon('../../common/wxwin.ico'))

        self._createControls()

        self._connectControls()

    def _createControls(self):
        # A Statusbar in the bottom of the window
        self.CreateStatusBar(1)
        sMsg = 'wxPython ' + wx.version()
        self.SetStatusText(sMsg)

        # Add a panel to the frame (needed under Windows to have a nice background)
        pnl = wx.Panel(self, wx.ID_ANY)

        szrMain = wx.BoxSizer(wx.VERTICAL)
        szrMain.AddStretchSpacer(1)
        self._btnShowMsg = wx.Button(pnl, wx.ID_ANY, 'Show a notification message')
        szrMain.Add(self._btnShowMsg, 0, wx.ALL|wx.ALIGN_CENTER, 5)
        szrMain.AddStretchSpacer(1)
        pnl.SetSizer(szrMain)

    def _connectControls(self):
        self._btnShowMsg.Bind(wx.EVT_BUTTON, self._onBtnShowMsgClicked)

    def _onBtnShowMsgClicked(self, evt):
        sTitle = 'wxWidgets notification'
        sMsg = 'This is a notification message.\n\nWelcome on wxWidgets ;-)'

        nmsg = wx.adv.NotificationMessage(title=sTitle, message=sMsg)
        nmsg.SetFlags(wx.ICON_INFORMATION)

        nmsg.Show(timeout=wx.adv.NotificationMessage.Timeout_Auto)
